iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -493,6 +493,8 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
                     OnTransportClosedGracefully();
 
                     await _closeRetryPolicy.ExecuteAsync(CleanUpImplAsync, cancellationToken).ConfigureAwait(true);
+                    var timeout = ((MqttTransportSettings)_transportSettings).GracefulEventLoopShutdownTimeout;
```

Comment:
Any unit tests you can add to validate?

Reply:
No unit tests, but I can add a E2E test that checks to see if the timeout is honored when disposing the object.